plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/ConnectionWrapper.java

```diff
@@ -204,9 +214,11 @@ private void updateWithResources(List<IProject> newValue) {
 			projects.clear();
 			for (IProject r : newValue) {
 				ProjectWrapper existingWrapper = oldWrappers.remove(r);
-
 				if (existingWrapper == null) {
 					ProjectWrapper newWrapper = new ProjectWrapper(this, r);
+					resourceCache.add(r);

```

Comment:
@jcantrill i understand that it's a rare case, but if we start watching on connection expansion, then if the project is deleted on the server just after creation (when we haven't expanded the connection yet), it would not be deleted in eclipse.
Jeff, i didn't understand about the project names. I provide it nowhere. Or what have i missed?